habitat-lab/habitat/config/default_structured_configs.py

```diff
@@ -1033,6 +1033,8 @@ class SimulatorConfig(HabitatBaseConfig):
     # ep_info is added to the config in some rearrange tasks inside
     # merge_sim_episode_with_object_config
     ep_info: Optional[Any] = None
+    # instance ids are recorded in the semantic sensor by adding obj_instance_id_start. This is added to avoid collision with scene semantic categories.
+    obj_instance_id_start: Optional[int] = 50
```

Comment:
This is for oracle experiments and we will most likely remove this later. This is used to provide a mask showing the objects that can be picked. The semantic sensor is set up to record instance ids. These are later matched to the valid instance ids (that come from episodes) for getting the mask. 

In general, I am setting up a more general version of the task we are solving. The category is only used as a prior (for goal specification). The episodes provide the ids of candidate instances to be picked up. This would allow us to use this code directly when we extend to complex language goals including object attributes. We will only need to update the episodes. Also, this keeps the complexity of mapping and checking for "category" away from the task.

If the name "semantic_sensor" makes this confusing, I can see if I can create a new instance id sensor.

Reply:
Yeah IMO, we shouldn't modify existing sensors if that is going to change the functionality of the sensor.

Reply:
Instance IDs seem useful to have, in general. Would this affect current downstream uses of the semantic sensor though? Could we just add a semantic + instance IDs sensor?

Reply:
I added Instance IDs to the semantic sensor space only for our rearrangement task. So, it should not affect other downstream applications. I think having a separate instance ID sensor is better. We needn't worry about conflicting output spaces.

Reply:
Keeping semantic + instance IDs as Chris suggested. (The sensor comes with GT semantics defined in the scene. I am adding the instance ids.) I am calling this a panoptic sensor now.

Reply:
sounds good